elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -629,12 +629,17 @@ public void bindEntity(Class<?> cls) {
      *
      * @param entityClass         Entity class type
      * @param annotationClassList List of sought annotations
+     * @param skipClassHierarchy Whether or not to include the class hierarchy in the search
      * @return annotation found
      */
     public static Annotation getFirstAnnotation(Class<?> entityClass,
-                                                List<Class<? extends Annotation>> annotationClassList) {
+                                                List<Class<? extends Annotation>> annotationClassList,
+                                                boolean skipClassHierarchy) {
         Annotation annotation = null;
         for (Class<?> cls = entityClass; annotation == null && cls != null; cls = cls.getSuperclass()) {
+            if (!cls.equals(entityClass) && skipClassHierarchy) {
```

Comment:
It is unnecessary to loop through remaining classes.  This could simply move down as
```java
if (skipClassHierarchy) {
   break;
}
```